File: ui/src/lib/types/post.ts
import type { User } from "./user_types";

export class Post {
    public id: string;
    public createdAt: Date;
    public updatedAt: Date;
    public caption: string;
    public images: string[];
    public poster: User;
    public location: string;

    constructor(id: string, createdAt: Date, updatedAt: Date, caption: string, images: string[], poster: User, location: string) {
        this.id = id;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.caption = caption;
        this.images = images;
        this.poster = poster;
        this.location = location;
    }
}

File: ent/schema/post.go
package schema

import (
	"entgo.io/ent"
	"entgo.io/ent/schema/edge"
	"entgo.io/ent/schema/field"
	"github.com/google/uuid"
	"time"
)

// Post holds the schema definition for the Post entity.
type Post struct {
	ent.Schema
}

// Fields of the Post.
func (Post) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.UUID{}),
		field.Time("created_at").Default(time.Now),
		field.Time("updated_at").Default(time.Now).UpdateDefault(time.Now),
		field.String("caption"),
		field.Strings("images"),
		field.String("location"),
	}
}

// Edges of the Post.
func (Post) Edges() []ent.Edge {
	return []ent.Edge{
		edge.From("poster", User.Type).Ref("posts").Unique(),
		edge.To("likes", Like.Type),
	}
}
